### PR TITLE
Reflect deprecation of .status.nodeRef in the NutanixMachine CRD

### PR DESCRIPTION
**Pull request by dlipovetsky**
**What this PR does / why we need it**:
In #331, we deprecated the `NutanixMachine.status.nodeRef` field, but it looks like the changes were not reflected in the CRD kustomization base:

```console
> git checkout 167cf7cf939bb182277f44765f995824c775d6cf
> make prepare-local-clusterctl
/home/dlipovetsky/nutanix/cluster-api-provider-nutanix/hack/tools/bin/controller-gen-v0.8.0 "crd:crdVersions=v1" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/home/dlipovetsky/nutanix/cluster-api-provider-nutanix/hack/tools/bin/kustomize-v4.5.4 build templates/base > templates/cluster-template.yaml
/home/dlipovetsky/nutanix/cluster-api-provider-nutanix/hack/tools/bin/kustomize-v4.5.4 build templates/csi > templates/cluster-template-csi.yaml
git mkdir -p ~/.cluster-api/overrides/infrastructure-nutanix/v1.3.99
cd config/manager && /home/dlipovetsky/nutanix/cluster-api-provider-nutanix/hack/tools/bin/kustomize-v4.5.4 edit set image controller=ghcr.io/nutanix-cloud-native/cluster-api-provider-nutanix/controller:latest
/home/dlipovetsky/nutanix/cluster-api-provider-nutanix/hack/tools/bin/kustomize-v4.5.4 build config/default > ~/.cluster-api/overrides/infrastructure-nutanix/v1.3.99/infrastructure-components.yaml
cp ./metadata.yaml ~/.cluster-api/overrides/infrastructure-nutanix/v1.3.99/
cp ./templates/cluster-template*.yaml ~/.cluster-api/overrides/infrastructure-nutanix/v1.3.99/
cp ./clusterctl.yaml ~/.cluster-api/clusterctl.yaml
> git diff --name-only
config/crd/bases/infrastructure.cluster.x-k8s.io_nutanixmachines.yaml
```

**Update:** I've added a make target to verify that generated manifests are up to date, and updated the `Test Build` GitHub Actions workflow to run this target.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**How Has This Been Tested?**:

**Update**: The updated `Test Build` workflow will verify that generated manifests are up to date. I'll paste an excerpt of the output, once the workflow runs.

```console
> make deploy
/home/dlipovetsky/nutanix/cluster-api-provider-nutanix/hack/tools/bin/controller-gen-v0.8.0 "crd:crdVersions=v1" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
GOOS=linux GOARCH=amd64 KO_DOCKER_REPO=ko.local /home/dlipovetsky/nutanix/cluster-api-provider-nutanix/hack/tools/bin/ko-v0.11.2 build -B -t latest -L .
2023/12/18 11:45:15 Using base gcr.io/distroless/static:nonroot@sha256:112a87f19e83c83711cc81ce8ed0b4d79acd65789682a6a272df57c4a0858534 for github.com/nutanix-cloud-native/cluster-api-provider-nutanix
2023/12/18 11:45:15 Building github.com/nutanix-cloud-native/cluster-api-provider-nutanix for linux/amd64
2023/12/18 11:45:18 Loading ko.local/cluster-api-provider-nutanix:d1d81bb3537e6a22921bead6b85ece34b0e622904e72cb040188984cf6422469
2023/12/18 11:45:20 Loaded ko.local/cluster-api-provider-nutanix:d1d81bb3537e6a22921bead6b85ece34b0e622904e72cb040188984cf6422469
2023/12/18 11:45:20 Adding tag latest
2023/12/18 11:45:20 Added tag latest
ko.local/cluster-api-provider-nutanix:d1d81bb3537e6a22921bead6b85ece34b0e622904e72cb040188984cf6422469
docker tag ko.local/cluster-api-provider-nutanix:latest ghcr.io/nutanix-cloud-native/cluster-api-provider-nutanix/controller:latest
kind load docker-image --name capi-test ghcr.io/nutanix-cloud-native/cluster-api-provider-nutanix/controller:latest
Image: "ghcr.io/nutanix-cloud-native/cluster-api-provider-nutanix/controller:latest" with ID "sha256:b73c301a3eb60d91c14c4194ce825f30ed73c9d964ccc993703986b15421f7e9" not yet present on node "capi-test-control-plane", loading...
clusterctl delete --infrastructure nutanix:v1.3.99 --include-crd || true
Using configuration File="/home/dlipovetsky/.cluster-api/clusterctl.yaml"
Deleting Provider="infrastructure-nutanix" Version="v1.3.99" Namespace="capx-system"
Deleting Endpoints="capx-controller-manager-metrics-service" Namespace="capx-system"
Deleting Secret="capx-nutanix-creds" Namespace="capx-system"
Deleting ConfigMap="capx-manager-config" Namespace="capx-system"
Deleting ConfigMap="capx-nutanix-endpoint" Namespace="capx-system"
Deleting ConfigMap="capx-user-ca-bundle" Namespace="capx-system"
Deleting Service="capx-controller-manager-metrics-service" Namespace="capx-system"
Deleting ServiceAccount="capx-controller-manager" Namespace="capx-system"
Deleting Deployment="capx-controller-manager" Namespace="capx-system"
Deleting Role="capx-leader-election-role" Namespace="capx-system"
Deleting ClusterRole="capx-system-capx-manager-role"
Deleting ClusterRole="capx-system-capx-metrics-reader"
Deleting ClusterRole="capx-system-capx-proxy-role"
Deleting ClusterRoleBinding="capx-system-capx-manager-rolebinding"
Deleting ClusterRoleBinding="capx-system-capx-proxy-rolebinding"
Deleting RoleBinding="capx-leader-election-rolebinding" Namespace="capx-system"
Deleting CustomResourceDefinition="nutanixclusters.infrastructure.cluster.x-k8s.io"
Deleting CustomResourceDefinition="nutanixmachines.infrastructure.cluster.x-k8s.io"
Deleting CustomResourceDefinition="nutanixmachinetemplates.infrastructure.cluster.x-k8s.io"
Deleting EndpointSlice="capx-controller-manager-metrics-service-jncnx" Namespace="capx-system"
Deleting Provider="infrastructure-nutanix" Namespace="capx-system"
Using configuration File="/home/dlipovetsky/.cluster-api/clusterctl.yaml"
clusterctl init --infrastructure nutanix:v1.3.99 -v 9
Using configuration File="/home/dlipovetsky/.cluster-api/clusterctl.yaml"
Fetching providers
Using Override="infrastructure-components.yaml" Provider="infrastructure-nutanix" Version="v1.3.99"
Fetching File="metadata.yaml" Provider="cluster-api" Type="CoreProvider" Version="v1.6.0"
Using Override="metadata.yaml" Provider="infrastructure-nutanix" Version="v1.3.99"
Creating Namespace="cert-manager-test"
Creating Issuer="test-selfsigned" Namespace="cert-manager-test"
Creating Certificate="selfsigned-cert" Namespace="cert-manager-test"
Deleting Namespace="cert-manager-test"
Deleting Issuer="test-selfsigned" Namespace="cert-manager-test"
Deleting Certificate="selfsigned-cert" Namespace="cert-manager-test"
Skipping installing cert-manager as it is already installed
Installing Provider="infrastructure-nutanix" Version="v1.3.99" TargetNamespace="capx-system"
Creating objects Provider="infrastructure-nutanix" Version="v1.3.99" TargetNamespace="capx-system"
Patching Namespace="capx-system"
Creating CustomResourceDefinition="nutanixclusters.infrastructure.cluster.x-k8s.io"
Creating CustomResourceDefinition="nutanixmachines.infrastructure.cluster.x-k8s.io"
Creating CustomResourceDefinition="nutanixmachinetemplates.infrastructure.cluster.x-k8s.io"
Creating ServiceAccount="capx-controller-manager" Namespace="capx-system"
Creating Role="capx-leader-election-role" Namespace="capx-system"
Creating ClusterRole="capx-system-capx-manager-role"
Creating ClusterRole="capx-system-capx-metrics-reader"
Creating ClusterRole="capx-system-capx-proxy-role"
Creating RoleBinding="capx-leader-election-rolebinding" Namespace="capx-system"
Creating ClusterRoleBinding="capx-system-capx-manager-rolebinding"
Creating ClusterRoleBinding="capx-system-capx-proxy-rolebinding"
Creating ConfigMap="capx-manager-config" Namespace="capx-system"
Creating ConfigMap="capx-nutanix-endpoint" Namespace="capx-system"
Creating ConfigMap="capx-user-ca-bundle" Namespace="capx-system"
Creating Secret="capx-nutanix-creds" Namespace="capx-system"
Creating Service="capx-controller-manager-metrics-service" Namespace="capx-system"
Creating Deployment="capx-controller-manager" Namespace="capx-system"
Creating inventory entry Provider="infrastructure-nutanix" Version="v1.3.99" TargetNamespace="capx-system"
Using configuration File="/home/dlipovetsky/.cluster-api/clusterctl.yaml"
# cd config/manager && /home/dlipovetsky/nutanix/cluster-api-provider-nutanix/hack/tools/bin/kustomize-v4.5.4 edit set image controller=ghcr.io/nutanix-cloud-native/cluster-api-provider-nutanix/controller:latest
# /home/dlipovetsky/nutanix/cluster-api-provider-nutanix/hack/tools/bin/kustomize-v4.5.4 build config/default | kubectl apply -f -
```

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and test output_

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by dlipovetsky** [comment]
I'll look at how we can automatically apply these changes on future PRs.

File: config/crd/bases/infrastructure.cluster.x-k8s.io_nutanixmachines.yaml [hunk 1]
```diff
@@ -309,8 +305,9 @@ spec:
                 description: Will be set in case of failure of Machine instance
                 type: string
               nodeRef:
-                description: NodeRef is a reference to the corresponding workload
-                  cluster Node if it exists.
+                description: 'NodeRef is a reference to the corresponding workload
```
Comment (deepakm-ntnx) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_nutanixmachines.yaml:308]: <pre>make manifests</pre> needs to be run to regenerate crd/bases

Comment (deepakm-ntnx) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_nutanixmachines.yaml:308]: assuming this was generated using that and not hand modified?

Comment (dlipovetsky) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_nutanixmachines.yaml:308]: Correct. `manifests` is a dependency of `prepare-local-clusterctl`, and I ran the latter in order to confirm the CRD deployment into the cluster.

**Comment by dlipovetsky** [comment]
> I'll look at how we can automatically apply these changes on future PRs.

I think we want to ensure that all manifest changes have been committed. We could do this by running `make manifest` as part of our PR checks. If the command changes the index (i.e. `git diff` is not empty), then the check should fail.

**Comment by dlipovetsky** [comment]
I've added a make target to verify the generated manifests are up to date, and changed the `Test Build` workflow to run this target. I'm putting these commits  _before_ the current commit, so I'm going to force-push.

**Review by dkoshkin — APPROVED** [review]
Thanks!

**Review by deepakm-ntnx — DISMISSED** [review]

**Comment by dlipovetsky** [comment]
I had to rebase on main. I did that and force-pushed. There were no merge conflicts.

**Comment by dlipovetsky** [comment]
I improved the message output by `make verify-manifests` when the manifests are out of date. Here's an example:

```console
> make verify-manifests
/home/dlipovetsky/nutanix/cluster-api-provider-nutanix/hack/tools/bin/controller-gen-v0.8.0 "crd:crdVersions=v1" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
diff --git a/config/crd/bases/infrastructure.cluster.x-k8s.io_nutanixmachines.yaml b/config/crd/bases/infrastructure.cluster.x-k8s.io_nutanixmachines.yaml
index c4ca53d..a15b74b 100644
--- a/config/crd/bases/infrastructure.cluster.x-k8s.io_nutanixmachines.yaml
+++ b/config/crd/bases/infrastructure.cluster.x-k8s.io_nutanixmachines.yaml
@@ -32,10 +32,6 @@ spec:
       jsonPath: .spec.providerID
       name: ProviderID
       type: string
-    - description: Corresponding workload cluster node
-      jsonPath: .status.nodeRef.name
-      name: NodeRef
-      type: string
     name: v1alpha4
     schema:
       openAPIV3Schema:
@@ -309,8 +305,9 @@ spec:
                 description: Will be set in case of failure of Machine instance
                 type: string
               nodeRef:
-                description: NodeRef is a reference to the corresponding workload
-                  cluster Node if it exists.
+                description: 'NodeRef is a reference to the corresponding workload
+                  cluster Node if it exists. Deprecated: Do not use. Will be removed
+                  in a future release.'
                 properties:
                   apiVersion:
                     description: API version of the referent.
@@ -370,10 +367,6 @@ spec:
       jsonPath: .spec.providerID
       name: ProviderID
       type: string
-    - description: Corresponding workload cluster node
-      jsonPath: .status.nodeRef.name
-      name: NodeRef
-      type: string
     name: v1beta1
     schema:
       openAPIV3Schema:
@@ -669,8 +662,9 @@ spec:
                 description: Will be set in case of failure of Machine instance
                 type: string
               nodeRef:
-                description: NodeRef is a reference to the corresponding workload
-                  cluster Node if it exists.
+                description: 'NodeRef is a reference to the corresponding workload
+                  cluster Node if it exists. Deprecated: Do not use. Will be removed
+                  in a future release.'
                 properties:
                   apiVersion:
                     description: API version of the referent.
generated manifests are out of date in the repository, run make manifests, and commit
make: *** [Makefile:606: verify-manifests] Error 1
```

**Comment by dlipovetsky** [comment]
Since GitHub required new reviews anyway, I squashed my fixup, and force-pushed.

**Review by thunderboltsid — APPROVED** [review]

**Review by adiantum — APPROVED** [review]
/lgtm
/approve
/woof

**Review by tuxtof — APPROVED** [review]

**Pull request merged**